src/core/server/http_resources/integration_tests/http_resources_service.test.ts

```diff
@@ -0,0 +1,224 @@
+        expect(response.header).toHaveProperty('content-type');
+        expect(response.header['content-type']).toBe('text/html; charset=utf-8');
+      });
+
+      it('renders javascript', async () => {
+        const { http, httpResources } = await root.setup();
+
+        const router = http.createRouter('');
+        const resources = httpResources.createRegistrar(router);
+        const jsBody = 'window.alert("from js body");';
+        resources.register({ path: '/render-js', validate: false }, (context, req, res) =>
+          res.renderJs({ body: jsBody })
+        );
+
+        await root.start();
+        const response = await kbnTestServer.request.get(root, '/render-js').expect(200);
+
+        expect(response.text).toBe(jsBody);
+        expect(response.header).toHaveProperty('content-type');
+        expect(response.header['content-type']).toBe('text/javascript; charset=utf-8');
+      });
+
+      it('attaches CSP header', async () => {
+        const { http, httpResources } = await root.setup();
+
+        const router = http.createRouter('');
+        const resources = httpResources.createRegistrar(router);
+        const htmlBody = `
+          <!DOCTYPE html>
+          <html>
+            <body>
+              <p>HTML body</p>
+            </body>
+          </html>
+        `;
+        resources.register({ path: '/render-html', validate: false }, (context, req, res) =>
+          res.renderHtml({ body: htmlBody })
+        );
+
+        await root.start();
+        const response = await kbnTestServer.request.get(root, '/render-html').expect(200);
+
+        expect(response.header).toHaveProperty('content-security-policy');
+        expect(response.header['content-security-policy']).toBe(defaultCspRules);
+      });
+
+      it('can attach headers, except the CSP & "content-type" headers', async () => {
+        const { http, httpResources } = await root.setup();
+
+        const router = http.createRouter('');
+        const resources = httpResources.createRegistrar(router);
+        resources.register({ path: '/render-core', validate: false }, (context, req, res) =>
+          res.renderHtml({
+            body: '<html><p>Hi</p></html>',
+            headers: {
+              'content-security-policy': "script-src 'unsafe-eval'",
+              'content-type': 'text/html',
+              'x-kibana': '42',
+            },
+          })
+        );
+
+        await root.start();
+        const response = await kbnTestServer.request.get(root, '/render-core').expect(200);
+
+        expect(response.header).toHaveProperty('content-security-policy');
+        expect(response.header['content-security-policy']).toBe(defaultCspRules);
+        expect(response.header['x-kibana']).toBe('42');
+      });
+
+      it('can adjust route config', async () => {
+        const { http, httpResources } = await root.setup();
+
+        const router = http.createRouter('');
+        const resources = httpResources.createRegistrar(router);
```

Comment:
I'm not happy that we have to duplicate the logic from `plugin_context` file, but it's simpler than adding a separate plugin for tests.